source/includes/cloudstack/_templates.md

```diff
@@ -110,8 +125,124 @@ Attributes | &nbsp;
 `dynamicallyScalable`<br/>*boolean* | true if you can dynamically scale an [instance](#cloudstack-instances) with this template
 `extractable`<br/>*boolean* | true if the template allows download from a generated URL
 `created`<br/>*string* | The creation date of the template
-`osTypeId`<br/>*UUID* | Id of th OS type
-`osType`<br/>*string* | The OS type of the template (e.g. Ubuntu, CentOS...)
+`osTypeId`<br/>*UUID* | Id of the OS type
+`osTypeName`<br/>*string* | The OS type of the template (e.g. Ubuntu, CentOS...)
 `zoneId`<br/>*UUID* |The [zone id](#cloudstack-zones) that the template is available in. Empty if available in multiple zones
 `zoneName`<br/>*string* |The [zone id](#cloudstack-zones) that the template is available in. Empty if available in multiple zones
 `availableInZones`<br/>*array[UUID]* | List of all [zone ids](#cloudstack-zones) that the template is available in
+`hypervisor`<br>*string* | The name of the hypervisor
+`format`<br>*string* | The template format for the chosen hypervisor
+`nicAdapter`<br>*string* | `VMWare only` The network interface controller name. Empty if not selected.<br>*Values*: e1000, pcnet32, vmxnet2 or vmxnet3
+`rootDiskController`<br>*string* | `VMWare only` The root disk controller name. Empty if not selected.<br>*Values*: ide, scsi for CloudStack version up to 4.4.<br>*Values*: ide, scsi, osdefault, pvscsi, lsilogic, lsisas1068, buslogic for CloudStack version 4.5 and up
+`keyboard`<br>*string* | `VMWare only` The keyboard type name. Empty if not selected.<br>*Values*: us, uk, jp, sc
+
+#### Import a template
+
+```shell
+curl -X POST \
+   -H "Content-Type: application/json" \
+   -H "MC-Api-Key: your_api_key" \
+   -d "request_body" \
+   "https://cloudmc_endpoint/v1/services/compute-on/test_area/templates"
+
+# Request should look like this
+```
+```json
+{
+  "name": "debian",
+  "description":"This is my template",
+  "url":"http://somewhere.com/template-vmware.ova",
+  "hypervisor":"VMWare",
+  "format":"OVA",
+  "keyboard":"us",
+  "nicAdapter":"vmxnet3",
+  "osTypeId": "53161a9c-6019-11e7-914a-0200246f00c5",
+  "osTypeName": "Debian GNU/Linux 7(64-bit)",
+  "zoneId":"e2cf7fa0-c3c2-4a39-9f8d-ee01ac3546cd"
+}
+```
+<code>POST /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/templates</code>
+
+Import a template
+
+Required | &nbsp;
+-------- | ------
+`name`<br/>*string* | The name of the template
+`description`<br/>*string* | The description of the template
+`url`<br/>*string* | The URL where the teplate is hosted. N.B. only `http` protocol is supported
```

Comment:
Is the NB true? Didn't we have a policy that dictated that?

Reply:
Disclaimer: this is copy/paste from above section.
But I will check non-the-less.

Let me know if you find anything else to be updated while I am at it.

Reply:
From the CS docs, v4.7:

url | the URL of where the template is hosted. Possible URL include http:// and https://

Reply:
description changes to mirror the CS documentation.